service/mapper/transitions_internal_test.go

```diff
@@ -136,771 +123,729 @@ func TestTransitions_BootstrapState(t *testing.T) {
+		tr, st := baselineFSM(t, StatusIndex)
+		tr.chain = chain
+		tr.write = write
 
-		err := tr.UpdateTree(st)
+		err := tr.IndexChain(st)
 
 		require.NoError(t, err)
-		assert.Equal(t, StatusMatched, st.status)
+		assert.Equal(t, StatusUpdate, st.status)
 	})
 
 	t.Run("handles invalid status", func(t *testing.T) {
 		t.Parallel()
 
-		tr, st := baselineFSM(t, StatusEmpty)
+		tr, st := baselineFSM(t, StatusBootstrap)
 
-		err := tr.UpdateTree(st)
+		err := tr.IndexChain(st)
 
 		assert.Error(t, err)
 	})
 
-	t.Run("handles feeder update failure", func(t *testing.T) {
+	t.Run("handles chain failure to retrieve commit", func(t *testing.T) {
 		t.Parallel()
 
-		feed := mocks.BaselineFeeder(t)
-		feed.UpdateFunc = func() (*ledger.TrieUpdate, error) {
-			return nil, mocks.GenericError
+		chain := mocks.BaselineChain(t)
+		chain.CommitFunc = func(uint64) (flow.StateCommitment, error) {
+			return flow.DummyStateCommitment, mocks.GenericError
 		}
 
-		tr, st := baselineFSM(t, StatusUpdating)
-		st.forest = mocks.BaselineForest(t, false)
-		tr.feed = feed
+		tr, st := baselineFSM(t, StatusIndex)
+		tr.chain = chain
 
-		err := tr.UpdateTree(st)
+		err := tr.IndexChain(st)
 
 		assert.Error(t, err)
 	})
 
-	t.Run("handles forest parent tree not found", func(t *testing.T) {
+	t.Run("handles writeer failure to write commit", func(t *testing.T) {
 		t.Parallel()
 
-		forest := mocks.BaselineForest(t, false)
-		forest.TreeFunc = func(_ flow.StateCommitment) (*trie.MTrie, bool) {
-			return nil, false
+		write := mocks.BaselineWriter(t)
+		write.CommitFunc = func(uint64, flow.StateCommitment) error {
+			return mocks.GenericError
 		}
 
-		tr, st := baselineFSM(t, StatusUpdating)
-		st.forest = forest
+		tr, st := baselineFSM(t, StatusIndex)
+		tr.write = write
 
-		err := tr.UpdateTree(st)
+		err := tr.IndexChain(st)
 
-		assert.NoError(t, err)
+		assert.Error(t, err)
 	})
-}
 
-func TestTransitions_CollectRegisters(t *testing.T) {
-	t.Run("nominal case", func(t *testing.T) {
+	t.Run("handles chain failure to retrieve header", func(t *testing.T) {
 		t.Parallel()
 
-		forest := mocks.BaselineForest(t, true)
-		forest.ParentFunc = func(commit flow.StateCommitment) (flow.StateCommitment, bool) {
-			assert.Equal(t, mocks.GenericCommit(0), commit)
-
-			return mocks.GenericCommit(1), true
+		chain := mocks.BaselineChain(t)
+		chain.HeaderFunc = func(uint64) (*flow.Header, error) {
+			return nil, mocks.GenericError
 		}
 
-		tr, st := baselineFSM(t, StatusMatched)
-		st.forest = forest
+		tr, st := baselineFSM(t, StatusIndex)
+		tr.chain = chain
 
-		err := tr.CollectRegisters(st)
+		err := tr.IndexChain(st)
 
-		require.NoError(t, err)
-		assert.Equal(t, StatusCollected, st.status)
-		for _, wantPath := range mocks.GenericLedgerPaths(6) {
-			assert.Contains(t, st.registers, wantPath)
-		}
+		assert.Error(t, err)
 	})
 
-	t.Run("indexing payloads disabled", func(t *testing.T) {
+	t.Run("handles writeer failure to write header", func(t *testing.T) {
```

Comment:
Few more occurrences below too.

```suggestion
	t.Run("handles writer failure to write header", func(t *testing.T) {
```